code/modules/power/singularity/emitter.dm

```diff
@@ -312,3 +356,148 @@
 		emagged = 1
 		if(user)
 			user.visible_message("[user.name] emags the [src.name].","<span class='notice'>You short out the lock.</span>")
+
+
+/obj/machinery/power/emitter/prototype
+	name = "Prototype Emitter"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "protoemitter"
+	icon_state_on = "protoemitter_+a"
+	can_buckle = TRUE
+	buckle_lying = 0
+	var/view_range = 12
+	var/datum/riding/riding_datum = null
+
+//BUCKLE HOOKS
+
+/obj/machinery/power/emitter/prototype/unbuckle_mob(mob/living/buckled_mob,force = 0)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	manual = FALSE
+	for(var/obj/item/I in buckled_mob.held_items)
+		if(istype(I, /obj/item/weapon/turret_control))
+			qdel(I)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = 0
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.change_view(world.view)
+	for(var/datum/action/innate/protoemitter/H in buckled_mob.actions)
+		qdel(H)

```

Comment:
Actions should be persistent. Instead of qdeling them and new()'ing them every time store one in the item then Grant() and Remove() it.